pkg/util/validation/limits.go

```diff
@@ -73,7 +76,172 @@ func (l *Limits) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	// We want to set c to the defaults and then overwrite it with the input.
 	// To make unmarshal fill the plain data struct rather than calling UnmarshalYAML
 	// again, we have to hide it using a type indirection.  See prometheus/config.
-	*l = defaultLimits
+	if defaultLimits != nil {
```

Comment:
Without this values were getting reset to 0 while loading cortex config which had just per tenant overrides config set.

Reply:
Why did this not happen before?

Reply:
@sandlis explained offline:
1. we init the flags, which sets up the default in a Limits struct in `cortex` pkg.
2. we then load YAML, which will overwrite them with 0s, as `defaultLimits` has not been inited.
3. we Parse the flags, which set only ones we've specifices
4. we then set this `defaultLimits` to the half-zero'd struct.